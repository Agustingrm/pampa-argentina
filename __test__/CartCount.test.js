import { render, screen } from "@testing-library/react";
import wait from "waait";
import CartCount from "../components/CartCount";

describe("<CartCount/>", () => {
  it("Renders", () => {
    render(<CartCount count={10} />);
  });
  it("Matches snapshot", () => {
    const { container } = render(<CartCount count={11} />);
    expect(container).toMatchSnapshot();
  });
  it("updates via props", async () => {
    const { container, rerender, debug } = render(<CartCount count={11} />);
    expect(container).toHaveTextContent("11");
    // Update the props
    rerender(<CartCount count="12" />);
    // First it concatenates two numbers (one shows up in top od the other in the page)
    expect(container).toHaveTextContent("1211");
    // wait for some ms, so the previous one dissapear
    await wait(400);
    expect(container).toHaveTextContent("12");
    expect(container).toMatchSnapshot();
  });
});
