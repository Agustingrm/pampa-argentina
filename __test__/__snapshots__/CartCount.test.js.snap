// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartCount/> Matches snapshot 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-1ampce4-1 YqugJ"
  >
    <div>
      <div
        class="CartCount__Dot-sc-1ampce4-0 jtRCi count"
      >
        11
      </div>
    </div>
  </span>
</div>
`;

exports[`<CartCount/> updates via props 1`] = `
<div>
  <span
    class="CartCount__AnimationStyles-sc-1ampce4-1 YqugJ"
  >
    <div>
      <div
        class="CartCount__Dot-sc-1ampce4-0 jtRCi count count-enter-done"
      >
        12
      </div>
    </div>
  </span>
</div>
`;
